RealmSwift/Projection.swift

```diff
@@ -0,0 +1,1342 @@
+public extension Projection {
+    /// :nodoc:
+    func addObserver(_ observer: NSObject,
+                     forKeyPath keyPath: String,
+                     options: NSKeyValueObservingOptions = [],
+                     context: UnsafeMutableRawPointer?) {
+        rootObject.addObserver(observer, forKeyPath: keyPath, options: options, context: context)
+    }
+
+    /// :nodoc:
+    func removeObserver(_ observer: NSObject,
+                        forKeyPath keyPath: String,
+                        context: UnsafeMutableRawPointer?) {
+        rootObject.removeObserver(observer, forKeyPath: keyPath, context: context)
+    }
+
+    /// :nodoc:
+    func removeObserver(_ observer: NSObject, forKeyPath keyPath: String) {
+        rootObject.removeObserver(observer, forKeyPath: keyPath)
+    }
+}
+
+// MARK: ThreadConfined
+extension Projection: ThreadConfined where Root: ThreadConfined {
+    /**
+     The Realm which manages the object, or `nil` if the object is unmanaged.
+     Note: Projection can be instantiated for the managed objects only therefore realm will never be nil.
+     Unmanaged objects are not confined to a thread and cannot be passed to methods expecting a
+     `ThreadConfined` object.
+     */
+    public var realm: Realm? {
+        rootObject.realm
+    }
+
+    /// Indicates if the object can no longer be accessed because it is now invalid.
+    public var isInvalidated: Bool {
+        return rootObject.isInvalidated
+    }
+    /**
+     Indicates if the object is frozen.
+     Frozen objects are not confined to their source thread. Forming a `ThreadSafeReference` to a
+     frozen object is allowed, but is unlikely to be useful.
+     */
+    public var isFrozen: Bool {
+        return realm?.isFrozen ?? false
+    }
+    /**
+     Returns a frozen snapshot of this object.
+     Unlike normal Realm live objects, the frozen copy can be read from any thread, and the values
+     read will never update to reflect new writes to the Realm. Frozen collections can be queried
+     like any other Realm collection. Frozen objects cannot be mutated, and cannot be observed for
+     change notifications.
+     Unmanaged Realm objects cannot be frozen.
+     - warning: Holding onto a frozen object for an extended period while performing write
+     transaction on the Realm may result in the Realm file growing to large sizes. See
+     `Realm.Configuration.maximumNumberOfActiveVersions` for more information.
+     */
+    public func freeze() -> Self {
+        let frozenObject = rootObject.freeze()
+        return Self(projecting: frozenObject)
+    }
+    /**
+     Returns a live (mutable) reference of this object.
+     Will return self if called on an already live object.
+     */
+    public func thaw() -> Self? {
+        if let thawedObject = rootObject.thaw() {
+            return Self(projecting: thawedObject)
+        }
+        return nil
+    }
+}
+
+// - MARK: _RealmSchemaDiscoverable
+extension Projection {
```

Comment:
This looks unnecessary?

Reply:
Conformance to `RealmCollectionValue` needs conformance to `_RealmSchemaDiscoverable`